HackIllinois/ViewControllers/HIGroupViewController.swift

```diff
@@ -0,0 +1,207 @@
+//
+//  HIGroupViewController.swift
+//  HackIllinois
+//
+//  Created by HackIllinois Team on 1/27/21.
+//  Copyright © 2021 HackIllinois. All rights reserved.
+//  This file is part of the Hackillinois iOS App.
+//  The Hackillinois iOS App is open source software, released under the University of
+//  Illinois/NCSA Open Source License. You should have received a copy of
+//  this license in a file with the distribution.
+//
+
+import Foundation
+import UIKit
+import CoreData
+
+class HIGroupViewController: HIGroupListViewController {
+
+    // MARK: - Properties
+    lazy var fetchedResultsController: NSFetchedResultsController<Profile> = {
+        let fetchRequest: NSFetchRequest<Profile> = Profile.fetchRequest()
+
+        fetchRequest.sortDescriptors = [
+            NSSortDescriptor(key: "firstName", ascending: true)
+        ]
+
+        let fetchedResultsController = NSFetchedResultsController(
+            fetchRequest: fetchRequest,
+            managedObjectContext: HICoreDataController.shared.viewContext,
+            sectionNameKeyPath: nil,
+            cacheName: nil
+        )
+
+        fetchedResultsController.delegate = self
+
+        return fetchedResultsController
+    }()
+
+    private var currentTab = 0
+    private var onlyFavorites = false
+    private let onlyFavoritesPredicate = NSPredicate(format: "favorite == YES" )
+    private let transparentBackground = HIView()
+    private let groupStatusTable = HITableView()
+    private var selectedButton = HIButton()
+    let horizontalStackView = UIStackView()
+    var buttonPresses = 0
+    private var interests = Set<String>()
+    private var selectedRows = Set<Int>()
+
+    @objc dynamic override func setUpBackgroundView() {
+            super.setUpBackgroundView()
+            backgroundView.image = #imageLiteral(resourceName: "GroupMatching")
+        }
+}
+
+// MARK: - UITabBarItem Setup
+extension HIGroupViewController {
+    override func setupTabBarItem() {
+        tabBarItem = UITabBarItem(title: "", image: #imageLiteral(resourceName: "matching"), tag: 0)
+    }
+}
+
+// MARK: - Actions
+extension HIGroupViewController {
+    func animateReload() {
+        try? fetchedResultsController.performFetch()
+        animateTableViewReload()
+        if let tableView = tableView, !tableView.visibleCells.isEmpty {
+            tableView.scrollToRow(at: IndexPath(row: 0, section: 0), at: .top, animated: false)
+        }
+    }
+    @objc func openPopup() {
+        let popupView = HIGroupPopupViewController()
+        popupView.modalPresentationStyle = .overCurrentContext
+        popupView.modalTransitionStyle = .crossDissolve
+        popupView.delegate = self
+        popupView.selectedRows = selectedRows
+        present(popupView, animated: true, completion: nil)
+    }
+
+    @objc func addDropdownView(button: HIButton) {
+        buttonPresses += 1
```

Comment:
popupActive = !popupActive

Reply:
this is a common way to make a toggle

Reply:
https://developer.apple.com/documentation/swift/bool/2994863-toggle